### PR TITLE
Exe 1626

### PR DESCRIPTION
**Pull request by ludvigla**
## Description

Changes name from `EdgeRankPlot` to `MoleculeRankPlot`. Keeping `EdgeRankPlot` for backwards compatibility, but it will be flagged as deprecated and it will dispatch to `MoleculeRankPlot` when run.

Fixes: exe-1626

## Type of change

Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [x] This change requires a documentation update.

## How Has This Been Tested?

Updated existing tests with new function name.

## PR checklist:

- [x] This comment contains a description of changes (with reason).
- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have checked my code and documentation and corrected any misspellings.
- [x] I have run R CMD check on the package and it passes without errors or warnings (notes can be acceptable if motivated)
- [x] I have documented any significant changes to the code in [CHANGELOG.md](../CHANGELOG.md)

### CODE REVIEW

**Review by maxkarlsson — APPROVED** [review]
Looks good! Saw a few instances of "edgerank" that I changed to "cellrank".

File: DESCRIPTION [hunk 1]
```diff
@@ -57,11 +57,11 @@ Collate:
     'CellGraphAssay.R'
     'aaa.R'
     'cell_count_plot.R'
+    'cell_rank_plot.R'
```
Comment (ambarrio) [inline DESCRIPTION:60]:
```suggestion
    'molecule_rank_plot.R'
```

**Review by ambarrio — CHANGES_REQUESTED** [review]
I would like to change the concept of `CellRankPlot` to `MoleculeRank?Plot` because as discussed some time in the past, `CellRankPlot` is not scalable to all kind of input we plan to have (e.g. cell interactions, organoids and tissues).

Let's take this up again for something that is scalable rather than changing back and forth.

File: DESCRIPTION (hunk 1 above)
Comment (ludvigla) [reply DESCRIPTION]: I have updated the function name to `MoleculeRankPlot` now!

Comment (ambarrio) [reply DESCRIPTION]: Nice! I approve

**Review by ambarrio — APPROVED** [review]

**Pull request merged**